-1 
Tested at: 696ffc9e381fb990ba071a7d191beda7c73c2215
 I found an error when building: 
 
 <pre>--- Registered EDM Plugin: testCaloSimHits
>> Building shared library tmp/slc6_amd64_gcc493/src/CalibCalorimetry/CaloTPG/src/CalibCalorimetryCaloTPG/libCalibCalorimetryCaloTPG.so
>> Building shared library tmp/slc6_amd64_gcc493/src/CalibCalorimetry/HcalAlgos/src/CalibCalorimetryHcalAlgos/libCalibCalorimetryHcalAlgos.so
--- Registered EDM Plugin: SimG4CMSCaloPlugins
/afs/cern.ch/cms/sw/ReleaseCandidates/vol1/slc6_amd64_gcc493/cms/cmssw-patch/CMSSW_8_0_X_2016-02-11-1100/external/slc6_amd64_gcc493/lib/libboost_date_time.so: file not recognized: Bad address
collect2: error: ld returned 1 exit status
gmake: *** [tmp/slc6_amd64_gcc493/src/CondTools/Geometry/plugins/CondToolsGeometryPlugins/libCondToolsGeometryPlugins.so] Error 1
>> Building shared library tmp/slc6_amd64_gcc493/src/CalibCalorimetry/CastorCalib/src/CalibCalorimetryCastorCalib/libCalibCalorimetryCastorCalib.so
Leaving library rule at src/SimG4CMS/ShowerLibraryProducer/plugins
@@@@ Running edmWriteConfigs for testCaloTowerGeometryESProducer
--- Registered EDM Plugin: testCaloTowerGeometryESProducer
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13255/11150/summary.html

--- a/Geometry/HcalCommonData/interface/HcalDDDRecConstants.h
+++ b/Geometry/HcalCommonData/interface/HcalDDDRecConstants.h
@@ -75,6 +75,7 @@ public:
   int                       getMaxDepth(const int type) const {return maxDepth[type];}
   int                       getMaxDepth(const int itype, const int ieta) const;
   int                       getNEta() const {return hpar->etagroup.size();}
+  int                       getNoff(const int i) const {return hpar->noff[i];}
   double                    getPhiBin(const int i) const {return phibin[i];}
   double                    getPhiOff(const int i) const {return hpar->phioff[i];}
   const std::vector<double> &      getPhiOffs()    const {return hpar->phioff;}

--- a/Geometry/HcalCommonData/src/HcalDDDRecConstants.cc
+++ b/Geometry/HcalCommonData/src/HcalDDDRecConstants.cc
@@ -6,7 +6,7 @@
 #include "CLHEP/Units/GlobalPhysicalConstants.h"
 #include "CLHEP/Units/GlobalSystemOfUnits.h"
 
-//#define DebugLog
+#define DebugLog
 
 enum { kHOSizePreLS1 = 2160, kHFSizePreLS1 = 1728 } ;
 
@@ -35,6 +35,9 @@ HcalDDDRecConstants::getEtaBins(const int itype) const {
   for (int ieta = iEtaMin[type]; ieta <= iEtaMax[type]; ++ieta) {
     int nfi = (int)((20.001*nModule[itype]*CLHEP::deg)/phibin[ieta-1]);
     HcalDDDRecConstants::HcalEtaBin etabin = HcalDDDRecConstants::HcalEtaBin(ieta, etaTable[ieta-1], etaTable[ieta], nfi, hpar->phioff[type], phibin[ieta-1]);
+    int n = (ieta == iEtaMax[type]) ? 0 : 1;
+    HcalDDDRecConstants::HcalEtaBin etabin0= HcalDDDRecConstants::HcalEtaBin(ieta, etaTable[ieta-1], etaTable[ieta+n], nfi, hpar->phioff[type], phibin[ieta-1]);
+    etabin0.depthStart = hcons.getDepthEta29(0)+1;
     int dstart = -1;
     if (layerGroupSize(ieta-1) > 0) {
       int lmin(0), lmax(0);
@@ -47,20 +50,26 @@ HcalDDDRecConstants::getEtaBins(const int itype) const {
 	  lmax = l + 1;
 	} else if ((int)layerGroup( ieta-1, l ) > dep) {
 	  if (dstart < 0) dstart = dep;
-	  etabin.layer.push_back(std::pair<int,int>(lmin,lmax));
+	  if (type == 1 && ieta+1 == hpar->noff[1] && dep > hcons.getDepthEta29(0)) {
+	    etabin0.layer.push_back(std::pair<int,int>(lmin,lmax));
+	  } else {
+	    etabin.layer.push_back(std::pair<int,int>(lmin,lmax));
+	  }
 	  lmin = (l + 1);
 	  lmax = l;
 	  dep  = layerGroup(ieta-1, l);
 	}
 	if (type == 0 && ieta == iEtaMax[type] && dep > hcons.getDepthEta16(0)) break;
+	if (type == 1 && ieta == hpar->noff[1] && dep > hcons.getDepthEta29(0)){
+	  lmax = lymx0;
+	  break;
+	}
       }
       if (lmax >= lmin) {
 	if (ieta+1 == hpar->noff[1]) {
-	} else if (ieta == hpar->noff[1]) {
-	  HcalDDDRecConstants::HcalEtaBin etabin0 = HcalDDDRecConstants::HcalEtaBin(ieta-1, etaTable[ieta-2], etaTable[ieta], nfi, hpar->phioff[type], phibin[ieta-1]);
-	  etabin0.depthStart = dep;
 	  etabin0.layer.push_back(std::pair<int,int>(lmin,lmax));
 	  bins.push_back(etabin0);
+	} else if (ieta == hpar->noff[1]) {
 	} else {
 	  etabin.layer.push_back(std::pair<int,int>(lmin,lmax));
 	  if (dstart < 0) dstart = dep;

--- a/Geometry/HcalCommonData/src/HcalDDDRecConstants.cc
+++ b/Geometry/HcalCommonData/src/HcalDDDRecConstants.cc
@@ -6,7 +6,7 @@
 #include "CLHEP/Units/GlobalPhysicalConstants.h"
 #include "CLHEP/Units/GlobalSystemOfUnits.h"
 
-#define DebugLog
+//#define DebugLog
 
 enum { kHOSizePreLS1 = 2160, kHFSizePreLS1 = 1728 } ;
 

